server/source_users.go

```diff
@@ -0,0 +1,317 @@
+package server
```

Comment:
Sometimes it is hard to review a git move unless then move is its own commit.  Many times I do the git move first as a commit and then do my other changes like getting the tests to work.  After a review I'll squash it together.

In any case, would you move this into the source.go file?  At some point we'll refactor source.go into its own package.